core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -215,12 +215,12 @@ object Controller {
 
   def start(args: Array[String])(implicit actorSystem: ActorSystem, logger: Logging): Unit = {
     ConfigMXBean.register()
-    Kamon.loadReportersFromConfig()
+    Kamon.init()
```

Comment:
Here we do not register the PrometheusReporter before init ... so does that mean for controller and invoker the prometheus reporter would open a new port?

Reply:
The `Kamon.loadModule` call should be done before this because the `MetricsRoute` is an object, and `KamonPrometheus` is evaluated immediately. In that sense, this should resemble the same behavior as before.

Reply:
For some reason with standalone atleast when I access http://localhost:3233/metrics then I do not see prometheus metrics being dumped. The output is empty.  Upon debugging I see metrics being sent to reporter but for some reason accumulator does not retain them. Not sure if its a bug in Kamon side

Reply:
Looks like a bug which was [recently fixed][1] (see kamon-io/kamon-prometheus#36). I see a 2.0.1 version of Prometheus reporter is now released with the fix

[1]: https://github.com/kamon-io/Kamon/issues/566#issuecomment-567835368